snapcraft/tests/integration/general/test_build_snaps.py

```diff
@@ -45,3 +47,26 @@ def test_build_snap(self):
         self.run_snapcraft('build')
         self.assertTrue(
             snapcraft.repo.snaps.SnapPackage.is_snap_installed(self.snap))
+
+
+class BuildSnapsErrorsTestCase(integration.TestCase):
+
+    def test_inexistent_build_snap(self):
```

Comment:
I'd also like to see a test for a snap that exists, but is unavailable in the given channel.

Reply:
oh, that is a much harder one :-)